bot/persistence/database_connector.py

```diff
@@ -0,0 +1,95 @@
+"""Contains logic for connecting to and manipulating the database.
+
+Todo:
+    * add queries for db tables once, the tables are clear
+"""
+
+from sqlite3 import Error
+from typing import List, Union
+from .queries import INSERT_PROPERTY_QUERY, GET_PROPERTY_QUERY
+from .database_manager import DatabaseManager
+
+
+class DatabaseConnector():
+    """Class used to communicate with the database.
+
+    The database is created and initialized with, using the __init__ method
+    The other methods support getting or adding properties to the database.
+    """
+
+    _db_file: Union[str, None] = None
+
+    def __init__(self, db_file: str, init_script=None):
+        """Create a database connection to a SQLite database and create the default tables form the SQL script in
+        init_db.sql.
+
+        Args:
+            db_file (str): The filename of the SQLite database file.
+            init_script (Union[str, None]): Optional SQL script filename that
+                will be run when the method is called.
+        """
+        self._db_file = db_file
+        with DatabaseManager(self._db_file) as db_manager:
```

Comment:
Out of curiosity: When does the `with` statement release resources? After it's done with its code or when our DatabaseConnector is destructed?